### PR TITLE
Add path to WritableFileWriter.

### PR DESCRIPTION
**Pull request by riversand963**
We want to sample the file I/O issued by RocksDB and report the function calls. This requires us to include the file paths otherwise it's hard to tell what has been going on.

Test plan:
```
$ make clean && make -j16 all check
```

### CODE REVIEW

**Comment by ajkr** [comment]
What is it for? Can we include the motivation for this change in the description?

File: tools/trace_analyzer_test.cc [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace rocksdb {
 namespace {
 static const int kMaxArgCount = 100;
 static const size_t kArgBufferSize = 100000;
-}
+}  // namespace
```
Comment (ajkr) [inline tools/trace_analyzer_test.cc]: did the linter add this? cool

Comment (riversand963) [reply tools/trace_analyzer_test.cc]: I don't remember -__-b
Thanks for your suggestion of `git add -p`

**Comment by riversand963** [comment]
Thanks @ajkr  for reviewing.

**Pull request closed without merging**